server/src/server.ts

```diff
@@ -73,6 +81,69 @@ connection.onInitialized(() => {
     }
 });
 
+function comparePositions(pos1: Position, pos2: Position) {
+    if (pos1.line !== pos2.line) {
+        return pos1.line > pos2.line ? 1 : -1;
+    }
+    if (pos1.character !== pos2.character) {
+        return pos1.character > pos2.character ? 1 : -1;
+    }
+    return 0;
+}
+
+function positionInRange(position: Position, range: Range) {
+    return comparePositions(range.start, position) <= 0 && comparePositions(position, range.end) <= 0;
+}
+
+
+
+connection.onDefinition((params) => {
+    const locations: Location[] = [];
+    const document = documents.get(params.textDocument.uri);
+    if (document) {
+        const links = getOmtDocumentResult(document);
+
+        links.calledObjects.forEach(link => {
+            if (positionInRange(params.position, link.range)) {
+                locations.push(...getLocationsForLink(params, links, link));
+            }
+        });
+    }
+
+    return locations;
+});
+
+function getLocationsForLink(params: DefinitionParams, links: OmtDocumentInformation, link: OmtLocalObject): Location[] {
+    const locations: Location[] = []
+    links.definedObjects.filter(x => x.name === link.name).forEach(t => {
+        locations.push(Location.create(params.textDocument.uri, t.range));
+    });
+    links.availableImports.filter(x => x.name === link.name).forEach(i => {
+        const linkUrl = `${i.fullUrl}`;
+        locations.push(...findDefinition(link, linkUrl));
+    });
+    return locations;
+}
+
+function findDefinition(link: OmtLocalObject, linkUrl: string): Location[] {
+    const locations: Location[] = [];
+    const otherDocument = TextDocument.create(linkUrl, 'omt', 1, fs.readFileSync(linkUrl).toString());
+    if (otherDocument) {
+        const result = omtDocumentInformationProvider.provideImportsFromDocument(otherDocument);
+        const definedObject = result.localDefinedObject.find(x => x.name == link.name);
```

Comment:
- [ ] Check if this function can be combined with `getLocationsForLink`

Reply:
I refactored this now so that the same function will be used for the current document and for other imported documents. We now have `getLocationsForLink` and `getLocationsInImportedFile` which loads the file and calls `getLocationsForLink`.